examples/pattern-matching/src/main/java/com/hazelcast/jet/examples/patternmatching/TransactionTracking.java

```diff
@@ -0,0 +1,139 @@
+
+    private static final String STATS_MAP_NAME = "transactionStatistics";
+    private static final long TIMED_OUT = -1L;
+    private static final int SLIDING_WINDOW_SIZE = 10;
+    private static final int SLIDING_STEP_MILLIS = 20;
+    private static final int EVENTS_PER_SECOND = 2_000;
+
+    private static Pipeline buildPipeline() {
+        Pipeline p = Pipeline.create();
+
+        // This stage emits pairs (transactionId, transactionDurationMillis).
+        // Transaction duration can also be the special value TIMED_OUT.
+        StreamStage<Entry<String, Long>> transactionOutcomes =
+                p.drawFrom(eventSource(EVENTS_PER_SECOND)).withTimestamps(TransactionEvent::timestamp, MAX_LAG)
+                 .groupingKey(TransactionEvent::transactionId)
+                 .mapStateful(
+                         SECONDS.toMillis(2),
+                         // This function creates the state object: an array of TransactionEvents.
+                         // The first slot is for the start event, the second for the end event:
+                         () -> new TransactionEvent[2],
+                         // This function takes the state, the grouping key, and the input item.
+                         // It updates the state and returns the output item (if any):
+                         (startEnd, transactionId, transactionEvent) -> {
+                             switch (transactionEvent.type()) {
+                                 case TRANSACTION_START:
+                                     startEnd[0] = transactionEvent;
+                                     break;
+                                 case TRANSACTION_END:
+                                     startEnd[1] = transactionEvent;
+                                     break;
+                                 default:
+                             }
+                             return (startEnd[0] != null && startEnd[1] != null)
+                                     ? entry(transactionId, startEnd[1].timestamp() - startEnd[0].timestamp())
+                                     : null;
+                         },
+                         // This function reacts to state eviction. It takes the state object being
+                         // evicted, the key associated with it, and the current watermark (i.e.,
+                         // current event time) and returns the output item, if any:
+                         (startEnd, transactionId, wm) -> (startEnd[0] == null || startEnd[1] == null)
+                                 ? entry(transactionId, TIMED_OUT)
+                                 : null
+                 );
+
+        // This pipeline branch computes the mean duration of transactions over a
+        // 10-second sliding window:
+        transactionOutcomes
+                .filter(e -> e.getValue() != TIMED_OUT)
+                .window(sliding(SECONDS.toMillis(SLIDING_WINDOW_SIZE), SLIDING_STEP_MILLIS))
```

Comment:
I think the windowing part and the graph didn't add much value for me. It would be maybe enough just to output the matched transactions and also the timed out ones?